packages/malloy/src/malloy.ts

```diff
@@ -2593,16 +2595,33 @@ class FluentState<T> {
  * queries or explores (via e.g. `loadFinalQuery()`, `loadQuery`, `loadExploreByName`, etc.).
  */
 export class ModelMaterializer extends FluentState<Model> {
+  private readonly replaceMaterializedReferences: boolean;
+  constructor(
+    protected runtime: Runtime,
+    materialize: () => Promise<Model>,
+    options?: PrepareResultOptions
+  ) {
+    super(runtime, materialize);
+    this.replaceMaterializedReferences =
+      options?.replaceMaterializedReferences ?? false;
+  }
+
   /**
    * Load the final (unnamed) Malloy query contained within this loaded `Model`.
    *
    * @return A `QueryMaterializer` capable of materializing the requested query, running it,
    * or loading further related objects.
    */
-  public loadFinalQuery(options?: PrepareQueryOptions): QueryMaterializer {
-    return this.makeQueryMaterializer(async () => {
-      return (await this.materialize()).getPreparedQuery(options);
-    });
+  public loadFinalQuery(options?: PrepareResultOptions): QueryMaterializer {
+    return this.makeQueryMaterializer(
+      async () => {
+        return (await this.materialize()).getPreparedQuery();
+      },
+      {
+        replaceMaterializedReferences: this.replaceMaterializedReferences,
+        ...options,
+      }
+    );
```

Comment:
Can say `modelMaterializer.loadFinalQuery({ replaceMaterializedReferences: true })`

Being able to do this this late might lead to some weird results... If you created a `ModelMaterializer` without passing in `{ replaceMaterializedReferences: true }`, then passed it in late, you would end up with something that had SQL blocks not use materialization, but queries using it:

```
# materialize
query: step_one is duckdb.table("foo") -> {
  select: a, b, c
}

# materialize
query: step_two is duckdb.sql("""
  SELECT * FROM (%{ step_one }) WHERE ...
""")

# materialize
query: step_three is step_two -> { ... }

run: step_three
```

In this case, you'd end up with `step_one` not being materialized.